src/app/api/[[...route]]/routes/rotation.route.ts

```diff
@@ -1,14 +1,88 @@
 import { Hono } from 'hono';
 
+import prisma from '@/lib/prisma';
+import type { IAssignedData } from '@/types/commons';
+
 const app = new Hono();
 
 export default app;
 
-app.get('/current/:shareHouseId', (c) => {
+app.get('/current/:shareHouseId', async (c) => {
   const shareHouseId = c.req.param('shareHouseId');
-  return c.json({
-    message: `Share house id for current rotation: ${shareHouseId}`,
-  });
+  try {
+    const shareHouseWithAssignmentSheet = await prisma.shareHouse.findUnique({
+      where: {
+        id: shareHouseId,
+      },
+      include: {
+        assignmentSheet: true,
+      },
+    });
+
+    if (
+      !shareHouseWithAssignmentSheet ||
+      !shareHouseWithAssignmentSheet.assignmentSheet
+    ) {
+      return c.json({ error: 'ShareHouse or AssignmentSheet not found' }, 404);
+    }
+
+    const assignedData = shareHouseWithAssignmentSheet.assignmentSheet
+      .assignedData as unknown as IAssignedData;
+
+    let totalTasks = 0;
+    let totalCompletedTasks = 0;
+
+    const categories = assignedData.assignments.map((assignment) => {
+      const maxTasks = assignment.tasks ? assignment.tasks.length : null;
+      const completedTasks = assignment.tasks
+        ? assignment.tasks.filter((task) => task.isCompleted === true).length
+        : null;
+
+      if (maxTasks !== null) totalTasks += maxTasks;
+      if (completedTasks !== null) totalCompletedTasks += completedTasks;
+
+      const categoryId = assignment.category ? assignment.category.id : null;
+      const categoryName = assignment.category
+        ? assignment.category.name
+        : null;
+
+      if (!assignment.tenant)
+        return c.json({ error: 'Internal Server Error' }, 500);
```

Comment:
Just like we discussed earlier, there's a potential if there's no tenant to be assigned to any category. Then, could you return data in a different format like below?

```ts
{
	name: string,
	startDate: string,
	endDate: string,
	progressRate: 0,
	categories: null
}
````

Reply:
I added an if statement to check whether all tenants are null.